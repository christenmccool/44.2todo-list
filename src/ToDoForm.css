.ToDoForm {
  margin: 10px 40px;
  border: 2px solid black;
  padding: 20px;
  background-color: lightgrey;
  border-radius: 5px;
}

.ToDoForm-group {
  display: flex;
  align-items: center;
}

.ToDoForm label {
  margin-right: 10px;
  font-size: 20px;
}

.ToDoForm input {
  padding: 5px;
  font-size: 20px;
  border-radius: 3px;
  width: 100%;
}

.ToDoForm select {
  padding: 5px;
  margin-left: 20px;
  font-size: 20px;
  border-radius: 3px;
}

.ToDoForm button {
  margin-top: 10px;
  padding: 5px 10px;
  margin-left: 20px;
  font-size: 20px;
  border-radius: 3px;
  margin-left: auto;
  color: white;
  background-color: black;
}