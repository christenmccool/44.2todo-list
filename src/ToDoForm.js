import React, {useState} from 'react';
import './ToDoForm.css';

const ToDoForm = ({addToDo}) => {
  const initialData = {
    task: "",
    priority: ""
  }
  const [formData, setFormData] = useState(initialData);

  const handleChange = (evt) => {
    const {name, value} = evt.target;
    setFormData({...formData, [name]:value});
  }

  const handleSubmit = (evt) => {
    evt.preventDefault();
    addToDo(formData);
    setFormData(initialData);
  }

  return (
    <div className="ToDoForm">
      <form className="ToDoForm-form" onSubmit={handleSubmit}>
        <div className="ToDoForm-group">
          <label htmlFor="task">Task:</label>
          <input id="task" name="task" value={formData.task} type="text" onChange={handleChange} required />
          {/* <label htmlFor="priority">Priority:</label> */}
          <select name="priority" id="priority" value={formData.priority} onChange={handleChange} required>
            <option value="" disabled selected>Select priority</option>
            <option value="low">Low</option>
            <option value="medium">Medium</option>
            <option value="high">High</option>
          </select>
        </div>
        <div className="ToDoForm-group">
          <button>Save</button>
        </div>
      </form>
    </div>
  )
}

export default ToDoForm;