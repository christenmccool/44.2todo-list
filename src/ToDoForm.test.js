import { render, asFragment, fireEvent } from '@testing-library/react';
import ToDoForm from './ToDoForm';


test('renders without crashing', () => {
  render(<ToDoForm />);
});

test('matches snapshot', () => {
  const {asFragment} = render(<ToDoForm />);    
  expect(asFragment()).toMatchSnapshot();
});

test('matches snapshot', () => {
  const addToDo = jest.fn();
  const {queryByLabelText, queryByText, debug} = render(<ToDoForm addToDo={addToDo}/>);    

  const task = queryByLabelText("Task:");
  const priority = queryByText("Select priority");
  const saveBtn = queryByText("Save");

  // fireEvent.change(task, {target: {value: 'Test task'}});
  // fireEvent.change(priority, {target: {value: 'medium'}});
  fireEvent.click(saveBtn);
  
  expect(addToDo).toHaveBeenCalledTimes(1);
});

