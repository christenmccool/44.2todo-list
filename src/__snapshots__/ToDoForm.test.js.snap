// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ToDoForm"
  >
    <form
      class="ToDoForm-form"
    >
      <div
        class="ToDoForm-group"
      >
        <label
          for="task"
        >
          Task:
        </label>
        <input
          id="task"
          name="task"
          required=""
          type="text"
          value=""
        />
        <select
          id="priority"
          name="priority"
          required=""
        >
          <option
            disabled=""
            value=""
          >
            Select priority
          </option>
          <option
            value="low"
          >
            Low
          </option>
          <option
            value="medium"
          >
            Medium
          </option>
          <option
            value="high"
          >
            High
          </option>
        </select>
      </div>
      <div
        class="ToDoForm-group"
      >
        <button>
          Save
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
